### PR TITLE
Raise minimum Swift version to 5.8

### PR DESCRIPTION
**Pull request by glbrntt**

### CODE REVIEW

**Comment by yim-lee** [comment]
CI for Swift 5.7 removed.

@swift-server-bot test this please

**Comment by yim-lee** [comment]
CI for Swift 5.7 removed

File: Benchmarks/Package.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version: 5.7
+// swift-tools-version: 5.8
```
Comment (yim-lee) [inline Benchmarks/Package.swift:1]: Could this change cause the benchmark failures?

```
14:21:46 Could not find any matching absolute thresholds at path [Thresholds/5.10/], failing threshold check.
```

**Comment by yim-lee** [comment]
Updated API breakage.

@swift-server-bot test this please

**Comment by Lukasa** [comment]
Seems like a benchmark regression here. @glbrntt are you able to look into this?

**Comment by glbrntt** [comment]
> Seems like a benchmark regression here. @glbrntt are you able to look into this?

Yes, I'll look into it.

**Pull request merged**